util/dataset.go

```diff
@@ -154,7 +159,18 @@ func (d *DataSet) Validate() error {
 		return fmt.Errorf("type is not valid: %s", d.Type)
 	}
 
-	if d.IngestPipeline == "" {
+	if d.Elasticsearch != nil && d.Elasticsearch.IngestPipelineName == "" {
```

Comment:
There is duplication of code inside this `if` block and it's sibling `else if` block. Looks like the intent is to prefer `d.Elasticsearch.IngestPipelineName` over `d.IngestPipeline`. How about defining a helper method, something like:

```
func (d *DataSet) ingestPipeline() string {
    if d.Elasticsearch != nil && d.Elasticsearch.IngestPipelineName != "" {
        return d.Elasticsearch.IngestPipelineName
    }

    return d.IngestPipeline
}
```

Then calling it here as using it's return value in the condition of the `if`. Then you should be able to get rid of the duplication.

Reply:
As the code goes away very soon, I kept it "delete friendly". Do you think it is still worth cleaning up?

Reply:
No, I was not aware it was going away soon when I made the clean up comment. Okay to leave it as-is.